File: utils/colors.py
"""
utils/colors.py — ANSI terminal color helpers
NuRichter · CySec Arsenal
"""

class C:
    RESET   = "\033[0m"
    BOLD    = "\033[1m"
    DIM     = "\033[2m"

    # NuRichter palette
    RED     = "\033[38;5;196m"
    ORANGE  = "\033[38;5;208m"
    YELLOW  = "\033[38;5;220m"
    GREEN   = "\033[38;5;82m"
    CYAN    = "\033[38;5;87m"
    BLUE    = "\033[38;5;63m"
    PURPLE  = "\033[38;5;135m"
    GRAY    = "\033[38;5;245m"
    WHITE   = "\033[97m"

    # Semantic
    OK      = GREEN
    WARN    = YELLOW
    ERR     = RED
    INFO    = CYAN
    FOUND   = ORANGE


def ok(msg: str)   -> str: return f"{C.OK}[+]{C.RESET} {msg}"
def warn(msg: str) -> str: return f"{C.WARN}[!]{C.RESET} {msg}"
def err(msg: str)  -> str: return f"{C.ERR}[-]{C.RESET} {msg}"
def info(msg: str) -> str: return f"{C.INFO}[*]{C.RESET} {msg}"
def found(msg: str)-> str: return f"{C.FOUND}[>]{C.RESET} {C.BOLD}{msg}{C.RESET}"

File: rust/arsenal-core/src/colors.rs
//! colors.rs — Terminal output helpers using `colored`
//! NuRichter · CySec Arsenal

use colored::Colorize;

/// `[+]` green — success
pub fn ok(msg: &str) -> String {
    format!("{} {}", "[+]".green().bold(), msg)
}

/// `[-]` red — error / not found
pub fn err(msg: &str) -> String {
    format!("{} {}", "[-]".red().bold(), msg)
}

/// `[!]` yellow — warning
pub fn warn(msg: &str) -> String {
    format!("{} {}", "[!]".yellow().bold(), msg)
}

/// `[*]` cyan — informational
pub fn info(msg: &str) -> String {
    format!("{} {}", "[*]".cyan().bold(), msg)
}

/// `[>]` orange/magenta — found something interesting
pub fn found(msg: &str) -> String {
    format!("{} {}", "[>]".bright_magenta().bold(), msg.bright_white().bold())
}

/// Separator line
pub fn sep(width: usize) -> String {
    "─".repeat(width).dimmed().to_string()
}

/// Print a labelled key-value row
pub fn kv(key: &str, val: &str) {
    println!("  {:<18} {}", key.bright_cyan(), val);
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn ok_contains_plus() {
        assert!(ok("test").contains("[+]"));
    }

    #[test]
    fn err_contains_minus() {
        assert!(err("test").contains("[-]"));
    }

    #[test]
    fn found_contains_arrow() {
        assert!(found("test").contains("[>]"));
    }
}
